File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        map<long long, int> count;
        int n = nums.size();
        vector<int> ans;
        for(int num: nums){
            count[num]++;
        }
        
        for(auto c: count){
            if (c.second > n/3 ){
                ans.push_back(c.first);
            }
        }
        
        return ans;
    }
};

File: 0229-majority-element-ii/0229-majority-element-ii.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
#         n = len(nums)
#         count = dict()
#         for num in nums:
#             if num not in count:
#                 count[num] =1
#             else:
#                 count[num] +=1
#         ans = []
#         for c in count.keys():
#             if count[c] > n//3:
#                 ans.append(c)
                
#         return ans
        
        # This is how we have to aim for in python
        # list to map with frequency using Counter function
        # Counter(nums).items() to iterate keys and values
        # condtion v i,e value if greatet than len(nums)// 3
        # taking k i,e key when condition is true
        
        return [ k for k, v in Counter(nums).items() if v > len(nums) // 3 ]
            
        